### PR TITLE
Run CI on nightly rust when proc-macro-* crates change

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

**Comment by lnicola** [comment]
I think this will bust the cache each time it switches toolchains. Maybe we can add the toolchain to the cache key.

**Comment by Veykril** [comment]
Good point

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -32,16 +32,21 @@ jobs:
...
+    needs: changes
     if: github.repository == 'rust-lang/rust-analyzer'
     name: Rust
     runs-on: ${{ matrix.os }}
     env:
       CC: deny_c
-      # we want to build r-a on stable to check that it keeps building on stable,
-      # but we also want to test our proc-macro-srv which depends on nightly features
-      RUSTC_BOOTSTRAP: 1
+      RUST_CHANNEL: "${{ needs.changes.outputs.proc_macros == 'true' && 'nightly' || 'stable'}}"
```
Comment (lnicola) [inline .github/workflows/ci.yaml:48]: Wow, I hate this syntax :-).

**Comment by Veykril** [comment]
Nice thing is that the rust-cross CI job will still compile on stable and therefore CI still checks that r-a builds on stable
@bors r+

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (Veykril) [reply .github/workflows/ci.yaml:48]: Same here, but it's the best I found :)

**Comment by Veykril** [comment]
Rust's bors really is quite chatty compared to bors-ng, at least I don't recall bors ng commenting so much

**Comment by Veykril** [comment]
oh right ... there was another issue here

**Comment by Veykril** [comment]
Okay there we go, nothing the good old bootstrap variable can't fix
@bors R+

**Pull request merged**